server/app/services/applicant/ApplicantService.java

```diff
@@ -428,33 +429,69 @@ private CompletionStage<ReadOnlyApplicantProgramService> stageAndUpdateIfValid(
    *     ApplicationSubmissionException} is thrown and wrapped in a `CompletionException`.
    */
   public CompletionStage<ApplicationModel> submitApplication(
-      long applicantId, long programId, CiviFormProfile submitterProfile, Request request) {
+      long applicantId, long programId, CiviFormProfile submitterProfile, Request request)
+      throws ProgramNotFoundException {
+    ProgramDefinition pd = programService.getFullProgramDefinition(programId);
     if (submitterProfile.isTrustedIntermediary()) {
       return getReadOnlyApplicantProgramService(applicantId, programId)
-          .thenCompose(ro -> validateApplicationForSubmission(ro, programId))
-          .thenComposeAsync(
-              v -> submitterProfile.getAccount(), classLoaderExecutionContext.current())
-          .thenComposeAsync(
-              tiAccount ->
-                  submitApplication(
-                      applicantId,
-                      programId,
-                      // /* tiSubmitterEmail= */
-                      // If the TI is submitting for themselves, don't set the tiSubmitterEmail. See
-                      // #5325 for more.
-                      tiAccount.ownedApplicantIds().contains(applicantId)
-                          ? Optional.empty()
-                          : Optional.of(tiAccount.getEmailAddress()),
-                      request),
-              classLoaderExecutionContext.current());
+          .thenCompose(
+              ro ->
+                  validateApplicationForSubmission(ro, programId)
+                      .thenComposeAsync(
+                          v -> submitterProfile.getAccount(), classLoaderExecutionContext.current())
+                      .thenComposeAsync(
+                          tiAccount -> {
+                            return getReadOnlyApplicantProgramService(applicantId, programId)
+                                .thenCompose(
+                                    ro2 -> {
+                                      EligibilityDetermination eligibilityDetermination =
+                                          Optional.of(pd.hasEligibilityEnabled())
```

Comment:
suggestion (non-blocking): The use of `Optional` here with the corresponding methods is clever but it made the code harder for me to understand. Since `hasEligibilityEnabled()` always returns a boolean (is never null), I would prefer a classic if/else check.